uaclient/sso.py

```diff
@@ -198,10 +200,13 @@ def discharge_root_macaroon(contract_client):
         caveat_id = extract_macaroon_caveat_id(root_macaroon['macaroon'])
         discharge_macaroon = prompt_request_macaroon(cfg, caveat_id)
     except (util.UrlError) as e:
-        logging.error("Could not reach url '%s' to authenticate.", e.url)
-        return None
+        msg = "Could not reach URL %s to authenticate"
+        logging.error(msg, e.url)
```

Comment:
It seems like in most cases we would want to be logging UserFactingErrors anyway, why aren't we perform the logging.error in cli.py:main_error_handler line 392?

Reply:
Great point; I'll look at addressing that.

Reply:
Yeah I think if you had the with disable_log_to_console context manager you could also drop it from the discharge_root_macaroon call site

Reply:
> Yeah I think if you had the with disable_log_to_console context manager in the exceptions.UserFacingError of main_error_handler you could also drop that ctx_mgr from the discharge_root_macaroon call site

Reply:
I've changed where the logging happens; I don't think we're yet ready to remove the context manager, though we might be by the end of this series of PRs.